code/modules/surgery/operations/operation_debride.dm

```diff
@@ -0,0 +1,83 @@
+/datum/surgery_operation/limb/debride
+	name = "debride infected flesh"
+	rnd_name = "Debridement"
+	desc = "Remove infected or necrotic flesh from a patient's wound to promote healing."
+	implements = list(
+		TOOL_HEMOSTAT = 1,
+		TOOL_SCALPEL = 1.25,
+		TOOL_SAW = 1.66,
+		TOOL_WIRECUTTER = 2.5,
+	)
+	time = 3 SECONDS
+	operation_flags = OPERATION_AFFECTS_MOOD | OPERATION_LOOPING | OPERATION_PRIORITY_NEXT_STEP
+	preop_sound = 'sound/items/handling/surgery/scalpel1.ogg'
+	success_sound = 'sound/items/handling/surgery/retractor2.ogg'
+	failure_sound = 'sound/items/handling/surgery/organ1.ogg'
+
+	/// How much infestation is removed per step (positive number)
+	var/infestation_removed = 4
+	/// How much sanitization is added per step
+	var/sanitization_added = 0.5 // just enough to stop infestation from worsening
+
+/datum/surgery_operation/limb/debride/all_required_strings()
+	return list("the limb must have a second degree or worse burn") + ..()
+
+/datum/surgery_operation/limb/debride/state_check(obj/item/bodypart/limb)
+	return !!(locate(/datum/wound/burn/flesh) in limb.wounds)
+
+/// To give the surgeon a heads up how much work they have ahead of them
+/datum/surgery_operation/limb/debride/proc/get_progress(datum/wound/burn/flesh/wound)
+	if(wound?.infestation <= 0)
+		return null
+
+	var/estimated_remaining_steps = wound.infestation / infestation_removed
```

Comment:
i would feel a lot less disgusted if it was `wound.infection` but thats out of scope